lib/money/money.rb

```diff
@@ -114,10 +114,13 @@ def -@
   end
 
   def <=>(other)
+    return unless other.respond_to?(:to_money)
+
     arithmetic(other) do |money|
-      value <=> money.value
+      value <=> money.value if currency.compatible?(money.currency)
```

Comment:
no need for `currency.compatible?(money.currency)`, this is already handled by `arithmetic(other) do`

Reply:
ah i see you want to make sure it's not equal instead of just having a deprecation. Ignore me

Reply:
AFAICT, `#arithmetic` doesn't guarantee that `money` will be in a matching currency? Likewise, 2 tests fail when removing the condition:

```
rspec ./spec/money_spec.rb:188 # Money does not eql? when currency missmatch
rspec ./spec/money_spec.rb:387 # Money frozen with amount of $1 <=> returns nil when comparing incompatible currency
```

So I think it has to stay for now? Maybe after `#arithmetic` raises on currency mismatch we'll be covered?